conda/activate.py

```diff
@@ -75,48 +72,45 @@ def get_export_unset_vars(self, odargs):
                        .. if you pass and set any other variable to None, then it
                        emits it to the dict with a value of None.
 
-        :return: A OrderedDict of env vars to export ordered the same way as kwargs.
+        :return: A dict of env vars to export ordered the same way as kwargs.
                  And a list of env vars to unset.
         """
-        kwargs = odargs
-        conda_exe_vars_None = False if ('conda_exe_vars' not in kwargs or
-                                        kwargs['conda_exe_vars'] is not None) else True
-        conda_exe_unset_vars = []
         unset_vars = []
-        # conda_exe_vars = context.conda_exe_vars_dict.copy()
-        from collections import OrderedDict
-        conda_exe_vars_export = OrderedDict()
-        for k, v in context.conda_exe_vars_dict.items():
-            if v is None or conda_exe_vars_None:
-                conda_exe_unset_vars.append(k)
-            else:
-                conda_exe_vars_export[k] = self.path_conversion(v) if v else v
-        from collections import OrderedDict
-        export_vars = OrderedDict()
-        for k, v in kwargs.items():
-            if k == 'conda_exe_vars':
-                continue
-            elif v is None:
-                unset_vars.append(k.upper())
-            else:
-                export_vars[k.upper()] = v
-        # Just to make writing tests a bit more ergonomic we add these to the end.
-        unset_vars += conda_exe_unset_vars
-        export_vars = OrderedDict(chain(export_vars.items(), conda_exe_vars_export.items()))
+        export_vars = {}
+
+        def split_export_unset(func=None, **kwargs):
+            for name, value in kwargs.items():
+                if value is None:
+                    unset_vars.append(name.upper())
+                elif func and value:
+                    export_vars[name.upper()] = func(value)
+                else:
+                    export_vars[name.upper()] = value
+
+        # split provided environment variables into exports vs unsets
+        split_export_unset(**kwargs)
+
+        if conda_exe_vars is None:
```

Comment:
I wonder if this is even necessary to have after having read this comment:

https://github.com/conda/conda/blob/6efd518f7b88d5682aa42b8b178d25a9e48bd1b2/conda/activate.py#L418-L425

It surely isn't being hit in our tests, so maybe if we do keep it, we could at least cover this line? Definitely a difficult decision to make when dealing with legacy code. I would err on the side of caution and keep it, but I would also add a new test case that at least reach this line, so that we know it works if used elsewhere.

Reply:
Hm, I hadn't fully processed this comment earlier.

This is related to the de-initialization discussions in https://github.com/conda/conda/issues/11087.

I'm inclined to "fix" the deactivate command in the future to unset all conda envvars after deactivating the last env (the only thing to remain would be `condabin` on the `PATH` which we could argue is also unnecessary) so this feature may become useful again in the future. Let's leave it for now.

**But** since it's an unused feature at the moment I have no problems (1) making it a positional argument (to better distinguish it from the other keyword args) and (2) renaming it as suggested above.